network-mux/demo/cardano-ping.hs

```diff
@@ -320,23 +326,26 @@ pingClient Options{quiet, magic, json} peer = bracket
            then return (msBlob sdu, t_e)
            else nextMsg bearer timeoutfn ptclNum
 
-    keepAlive :: MuxBearer IO -> TimeoutFn IO -> String -> TDigest 5 -> Word16 -> IO ()
+    keepAlive :: MuxBearer IO -> TimeoutFn IO -> String -> TDigest 5 -> Word32 -> IO ()
+    keepAlive _ _ _ _ cookie | cookie == maxCount = return ()
     keepAlive bearer timeoutfn peerStr td cookie = do
-        !t_s <- write bearer timeoutfn $ wrap keepaliveNum InitiatorDir (keepAliveReq cookie)
+        when (cookie > 0 && json) $ putStrLn ","
```

Comment:
This is not very elegant, but I don't have an idea how to make it nicer.

Reply:
It also doesn't work when we're pinging multiple hosts. Will think of another way.